localstack/services/transcribe/provider.py

```diff
@@ -208,6 +211,26 @@ def _run_transcription_job(self, args: Tuple[TranscribeStore, str]):
             bucket, _, key = s3_path.removeprefix("s3://").partition("/")
             s3_client.download_file(Bucket=bucket, Key=key, Filename=file_path)
 
+            # Check audio file type and convert to wav
+            support_audio_list = [
+                "audio/flac",
+                "audio/mpeg",
+                "video/mp4",
+                "audio/ogg",
+                "video/webm",
+            ]
+            file_type = magic.from_file(file_path, mime=True)
+            if file_type != "audio/x-wav" and file_type in support_audio_list:
+                LOG.debug("Starting to convert audio to wav type: %s", job_name)
+
+                tmp_wav_file_path = new_tmp_file() + ".wav"
+                cmd = "ffmpeg -nostdin -loglevel quiet -y -i {} {}".format(
+                    str(file_path), tmp_wav_file_path
+                )
+                subprocess.run(cmd.split(" "), stdout=subprocess.PIPE, check=True)
+                file_path = tmp_wav_file_path
+                LOG.debug("Finishing conversion: %s", job_name)
+
```

Comment:
What if the conversion fails?
Scenario 1: User provides an invalid file
-> ffmpeg fails silently and no file is generated. `audio = wave.open(file_path, "rb")` will raise a `FileNotFoundError`.
this is ok since it's all in a try-except block, but could use a better `failure_reason`

Scenario 2: conversion process completes, but the generated file still does not have the right format.
-> the `failure_reason` would be "Audio file must be mono PCM WAV format", which is not helpful. Can you add a more helpful message to debug?

Reply:
Leaving this out as a future `TODO`, based on how much this the transcribe service is used and its value.